mathics/builtin/patterns/rules.py

```diff
@@ -0,0 +1,563 @@
+
+        all_list = all(rule.has_form("List", None) for rule in rules)
+        if all_list:
+            elements = [self.eval_create(rule, evaluation) for rule in rules]
+            return ListExpression(*elements)
+        flatten_list = []
+        for rule in rules:
+            if isinstance(rule, Symbol):
+                rule = rule.evaluate(evaluation)
+            if rule.has_form("List", None):
+                flatten_list.extend(rule.elements)
+            elif rule.has_form(("Rule", "RuleDelayed"), 2):
+                flatten_list.append(rule)
+            elif isinstance(rule, Dispatch):
+                flatten_list.extend(rule.src.elements)
+            else:
+                # WMA does not raise this message: just leave it unevaluated,
+                # and raise an error when the dispatch rule is used.
+                evaluation.message("Dispatch", "invrpl", rule)
+                return None
+        try:
+            return Dispatch(flatten_list, evaluation)
+        except Exception:
+            return None
+
+    def eval_normal(self, dispatch: Dispatch, evaluation: Evaluation) -> ListExpression:
+        """Normal[dispatch_Dispatch]"""
+        if isinstance(dispatch, Dispatch):
+            return dispatch.src
+        return dispatch.elements[0]
+
+
+class Replace(Builtin):
+    """
+
+    <url>
+    :WMA link:
+    https://reference.wolfram.com/language/ref/Replace.html</url>
+
+    <dl>
+      <dt>'Replace[$expr$, $x$ -> $y$]'
+      <dd>yields the result of replacing $expr$ with $y$ if it
+        matches the pattern $x$.
+      <dt>'Replace[$expr$, $x$ -> $y$, $levelspec$]'
+      <dd>replaces only subexpressions at levels specified through
+        $levelspec$.
+      <dt>'Replace[$expr$, {$x$ -> $y$, ...}]'
+      <dd>performs replacement with multiple rules, yielding a
+        single result expression.
+      <dt>'Replace[$expr$, {{$a$ -> $b$, ...}, {$c$ -> $d$, ...}, ...}]'
+      <dd>returns a list containing the result of performing each
+        set of replacements.
+    </dl>
+
+    >> Replace[x, {x -> 2}]
+     = 2
+
+    By default, only the top level is searched for matches
+    >> Replace[1 + x, {x -> 2}]
+     = 1 + x
+
+    >> Replace[x, {{x -> 1}, {x -> 2}}]
+     = {1, 2}
+
+    Replace stops after the first replacement
+    >> Replace[x, {x -> {}, _List -> y}]
+     = {}
+
+    Replace replaces the deepest levels first
+    >> Replace[x[1], {x[1] -> y, 1 -> 2}, All]
+     = x[2]
+
+    By default, heads are not replaced
+    >> Replace[x[x[y]], x -> z, All]
+     = x[x[y]]
+
+    Heads can be replaced using the Heads option
+    >> Replace[x[x[y]], x -> z, All, Heads -> True]
+     = z[z[y]]
+
+    Note that heads are handled at the level of elements
+    >> Replace[x[x[y]], x -> z, {1}, Heads -> True]
+     = z[x[y]]
+
+    You can use Replace as an operator
+    >> Replace[{x_ -> x + 1}][10]
+     = 11
+    """
+
+    messages = {
+        "reps": "`1` is not a valid replacement rule.",
```

Comment:
This line can be deleted if should have been "rep". And then the message call below in 255 should be adjusted.

Reply:
BTW I haven't been able to check this since I can't get to galecia

Reply:
> BTW I haven't been able to check this since I can't get to galecia

Yes, it seems there is a power outage in the building where galatea is located.